src/Symfony/Component/HttpKernel/Controller/ArgumentResolver.php

```diff
@@ -85,16 +96,16 @@ public function getArguments(Request $request, callable $controller, \Reflection
     }
 
     /**
-     * @return iterable<int, ArgumentValueResolverInterface>
+     * @return iterable<string, ArgumentValueResolverInterface>
```

Comment:
Can I do that or should I reindex resolvers in the constructor?

Reply:
this doesn't work yes: this hardcodes the names that we also declare in the DI config. But nothing prevents registering those under a different name in theory. I think this hints this might not be the right approach.
I don't think ArgumentResolver should be patched to achieve what we want with this PR.
Instead, I stick with my previous proposal: PinnedResolverValueResolver should not choke when the pinned resolver yield nothing. This would make all this unnecessary (at the cost of requiring implementations to be more careful, but that's OK IMHO)